File: src/test/java/com/raychenon/leetcode/RomanToIntegerTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-09-13
 * https://leetcode.com/problems/roman-to-integer
 */
public class RomanToIntegerTest {

    private static RomanToInteger romanToInteger;

    @BeforeClass
    public static void setUp() {
        romanToInteger = new RomanToInteger();
    }

    @Test
    public void testIII() {
        assertEquals(3, romanToInteger.romanToInt("III"));
    }

    @Test
    public void testXIV() {
        assertEquals(14, romanToInteger.romanToInt("XIV"));
    }

    @Test
    public void testLVIII() {
        assertEquals(58, romanToInteger.romanToInt("LVIII"));
    }

    @Test
    public void testMCMXCIV() {
        assertEquals(1994, romanToInteger.romanToInt("MCMXCIV"));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/RomanToIntegerTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-09-13
 * https://leetcode.com/problems/roman-to-integer/
 */
class RomanToIntegerTest {

    @Test
    fun testIII() {
        Assert.assertEquals(3, RomanToInteger.romanToInt("III"))
    }

    @Test
    fun testIV() {
        Assert.assertEquals(4, RomanToInteger.romanToInt("IV"))
    }

    @Test
    fun testIX() {
        Assert.assertEquals(9, RomanToInteger.romanToInt("IX"))
    }

    @Test
    fun testLVIII() {
        Assert.assertEquals(58, RomanToInteger.romanToInt("LVIII"))
    }

    @Test
    fun testMCMXCIV() {
        Assert.assertEquals(1994, RomanToInteger.romanToInt("MCMXCIV"))
    }
}